lib/std/crypto/25519/ed25519.zig

```diff
@@ -0,0 +1,127 @@
+const std = @import("std");
+const fmt = std.fmt;
+const mem = std.mem;
+const Sha512 = std.crypto.Sha512;
+
+/// Ed25519 (EdDSA) signatures.
+pub const Ed25519 = struct {
+    /// The underlying elliptic curve.
+    pub const Curve = @import("edwards25519.zig").Edwards25519;
+    /// Length (in bytes) of a seed required to create a key pair.
+    pub const seed_length = 32;
+    /// Length (in bytes) of a compressed key pair.
+    pub const keypair_length = 64;
+    /// Length (in bytes) of a compressed public key.
+    pub const public_length = 32;
+    /// Length (in bytes) of a signature.
+    pub const signature_length = 64;
+    /// Length (in bytes) of optional random bytes, for non-deterministic signatures.
+    pub const noise_length = 32;
+
+    /// Derive a key pair from a secret seed.
+    pub fn createKeyPair(seed: [seed_length]u8) ![keypair_length]u8 {
+        var az: [Sha512.digest_length]u8 = undefined;
+        var h = Sha512.init();
```

Comment:
Or a `rand` interface?

Reply:
Given that @jedisct1 intends to refactor our error values into a common error enum set in a subsequent PR, perhaps this could be done in another PR?

Reply:
Ed25519 has always been defined with SHA512, from NaCl to RFC8032. I'm not sure we want to let users invent their own construction, especially since the AMAC guarantees may not hold any more.

Reply:
> Given that @jedisct1 intends to refactor our error values into a common error enum set in a subsequent PR, perhaps this could be done in another PR?

Sure; I left my feedback as a "comment" rather than a "request changes" to indicate that I'm fine with merging this code now and having a followup PR. We don't have to get things perfect on the first try.

> Ed25519 has always been defined with SHA512, from NaCl to RFC8032.

I didn't know this :) Sounds reasonable to me to keep Sha512 here then. Maybe add something (doc)comment that it's not just an arbitrary choice.

Reply:
Comment added.